src/main/java/io/jenkins/plugins/explain_error/ErrorExplainer.java

```diff
@@ -70,11 +70,17 @@ private String extractErrorLogs(Run<?, ?> run, String logPattern, int maxLines)
         Pattern pattern = Pattern.compile(logPattern, Pattern.CASE_INSENSITIVE);
         StringBuilder errorLogs = new StringBuilder();
 
-        for (String line : logLines) {
-            if (pattern.matcher(line).find()) {
-                errorLogs.append(line).append("\n");
-            }
-        }
+for (int i = 0; i < logLines.size(); i++) {
+    String line = logLines.get(i);
+
+    if (pattern.matcher(line).find()) {
+        int lineNumber = i + 1; // 1 indexing
+        System.out.println("Matched error at line: " + lineNumber);
```

Comment:
Never use `System.out` or `System.err` in Jenkins. If you want to write to the Jenkins log use the `LOGGER`